qiskit/transpiler/passes/synthesis/high_level_synthesis_plugins.py

```diff
@@ -0,0 +1,93 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Default plugins for synthesizing high-level-objects in Qiskit.
```

Comment:
Can this file be part of `qiskit/transpiler/passes/synthesis/plugin.py`?

Reply:
This also makes sense, especially if we want to generalize the documentation to cover both cases. Though let me redirect this question to @mtreinish, to see if he agrees with this.

Reply:
I agree with this idea, we should integrate the docs into one module docstring. I think the details are sufficiently different we'll want 2 sections for each interface, mostly because implementing the plugins are different. But the pieces about using entry points to expose plugins are sufficiently the same we can abstract that if you think it's best. FWIW, this is why the section hierarchy has the top header as `Writing Plugins` and the subsection is `Unitary Synthesis Plugins`. My thinking at the time was we could easily add any other synthesis interfaces to that list with a new subsection. So in this case add `High Level Synthesis` as a subsection there

Reply:
Done (see the previous comment for more details).